lib/storage/providers/AsyncStorage.js

```diff
@@ -56,8 +61,35 @@ const provider = {
      * @return {Promise<void>}
      */
     multiMerge(pairs) {
-        const stringPairs = _.map(pairs, ([key, value]) => [key, JSON.stringify(value)]);
-        return AsyncStorage.multiMerge(stringPairs);
+        const tasks = _.map(pairs, ([key, value, currentValue]) => {
+            if (mergeQueue[key]) {
+                mergeQueue[key].push(value);
+                return Promise.resolve();
```

Comment:
This method sometimes resolve early, when the write is not yet complete

---

I think we can skip the changes here and delete this storage provider 
IIRC it's no longer used
- web/desktop uses LocalForage
- native uses SQLite

Reply:
It's still being used for tests though.

Reply:
I'm currently trying to remove all the remaining dependencies on async-storage in the tests. After that, we should be ready to remove it